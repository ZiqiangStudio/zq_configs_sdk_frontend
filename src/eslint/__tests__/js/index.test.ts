import { curryExecEslint } from '../util';

describe('JS Lint', () => {
  const execEslint = curryExecEslint('js');

  test('报错', async () => {
    const { output, error } = await execEslint('error.cjs');
    expect(error).toBeTruthy();
    expect(output).toMatchSnapshot();
  });

  test('正常通过', async () => {
    const { output, error } = await execEslint('normal.cjs');
    expect(error).toBeFalsy();
    expect(output).toMatchSnapshot();
  });
});
