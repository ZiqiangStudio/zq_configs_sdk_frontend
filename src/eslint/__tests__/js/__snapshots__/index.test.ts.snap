// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JS Lint 报错 1`] = `" /js/error.cjs    1:66  error  Replace \`"../config")\` with \`'../config');\`                                                 prettier/prettier    5:21  error  Insert \`;\`                                                                                  prettier/prettier    6:50  error  Insert \`;\`                                                                                  prettier/prettier    7:53  error  Insert \`;\`                                                                                  prettier/prettier    9:23  error  Insert \`;\`                                                                                  prettier/prettier   10:21  error  Insert \`;\`                                                                                  prettier/prettier   12:31  error  Insert \`;\`                                                                                  prettier/prettier   18:45  error  Insert \`;\`                                                                                  prettier/prettier   22:6   error  Insert \`;\`                                                                                  prettier/prettier   23:5   error  Remove this conditional structure or edit its code blocks so that they're not all the same  sonarjs/no-all-duplicated-branches   23:41  error  Insert \`;\`                                                                                  prettier/prettier   24:31  error  Insert \`;\`                                                                                  prettier/prettier   25:58  error  Insert \`;\`                                                                                  prettier/prettier   26:9   error  Unexpected constant condition                                                               no-constant-condition   26:29  error  Insert \`;\`                                                                                  prettier/prettier   27:16  error  Insert \`;\`                                                                                  prettier/prettier   31:29  error  Insert \`;\`                                                                                  prettier/prettier   32:31  error  Insert \`;\`                                                                                  prettier/prettier   36:27  error  Insert \`;\`                                                                                  prettier/prettier   37:31  error  Insert \`;\`                                                                                  prettier/prettier   41:35  error  Insert \`;\`                                                                                  prettier/prettier   42:31  error  Insert \`;\`                                                                                  prettier/prettier   46:24  error  Insert \`;\`                                                                                  prettier/prettier  ✖ 23 problems (23 errors, 0 warnings)   21 errors and 0 warnings potentially fixable with the \`--fix\` option.  "`;

exports[`JS Lint 正常通过 1`] = `""`;
